src/test/user.spec.js

```diff
@@ -674,3 +674,140 @@ describe('PATCH /api/v1/auth/user/:id/role', () => {
+
+  it('should get all followers', async () => {
+    const firstUser = await createUser();
+    const secondUser = await createUser();
+    const thirdUser = await createUser();
+
+    await BaseRepository.create(db.Follower, {
+      followeeId: firstUser.id,
+      followerId: secondUser.id
+    });
+    await BaseRepository.create(db.Follower, {
+      followeeId: firstUser.id,
+      followerId: thirdUser.id
+    });
+
+    const token = helper.jwtSigner(firstUser);
+
+    const numberOfFollowing = await BaseRepository.findAndCountAll(
+      db.Follower,
+      {
+        followerId: firstUser.id
+      }
+    );
+
+    expect(numberOfFollowing.count).to.equal(2);
+
+    const page = 1;
+    const limit = 2;
+
+    const res = await server()
+      .get(`${USER_API}/followers?page=${page}&limit=${limit}`)
+      .set('token', token)
+      .send({ followeeId: firstUser.id });
+    expect(res.status).to.equal(200);
+    expect(res.body.message.length).to.equal(numberOfFollowing.count);
+  });
+});
+
+describe('GET /api/v1/users/followings', () => {
+  beforeEach(async () => {
+    await db.User.destroy({ cascade: true, truncate: true });
+    await db.Follower.destroy({ cascade: true, truncate: true });
+  });
+
+  it('returns appropriate message when user is not following anyone', async () => {
+    const firstUser = await createUser();
+
+    const token = helper.jwtSigner(firstUser);
+
+    const numberOfFollowing = await BaseRepository.findAndCountAll(
+      db.Follower,
+      {
+        followeeId: firstUser.id
+      }
+    );
+
+    expect(numberOfFollowing.count).to.equal(0);
+    const limit = 1;
+    const page = 1;
+
+    const res = await server()
+      .get(`${USER_API}/following?page=${page}&limit=${limit}`)
+      .set('token', token)
+      .send({ followeeId: firstUser.id });
+    expect(res.status).to.equal(200);
+    expect(res.body.message).to.equal(
+      `You are not following anyone at the moment`
+    );
+  });
+
+  it('should get all followers', async () => {
+    const firstUser = await createUser();
+    const secondUser = await createUser();
+    const thirdUser = await createUser();
+
+    await BaseRepository.create(db.Follower, {
+      followeeId: firstUser.id,
+      followerId: secondUser.id
+    });
+    await BaseRepository.create(db.Follower, {
+      followeeId: firstUser.id,
+      followerId: thirdUser.id
+    });
+
+    const token = helper.jwtSigner(firstUser);
+
+    const numberOfFollowing = await BaseRepository.findAndCountAll(
+      db.Follower,
+      {
+        followerId: firstUser.id
+      }
+    );
+
+    expect(numberOfFollowing.count).to.equal(2);
+    const limit = 2;
+    const page = 1;
+
+    const res = await server()
+      .get(`${USER_API}/followers?page=${page}&limit=${limit}`)
+      .set('token', token)
+      .send({ followeeId: firstUser.id });
+    expect(res.status).to.equal(200);
+    expect(res.body.message.length).to.equal(numberOfFollowing.count);
+  });
+});
```

Comment:
I can see you added pagination on this page. So you need to test the response of the pagination returned on that page. You may need to create more followers so you can assert for the `prev page`, `currentPage`, `next page` and `total number of pages`. You can check the test on [fetch all articles page](https://github.com/andela/ah-kifaru-backend/blob/develop/src/test/article.spec.js). It should be of help.

Reply:
Wow, thank you so much for this @Onyimatics , I would make the changes soon.